metriq_gym/run.py

```diff
@@ -112,11 +117,61 @@ def poll_job(args: argparse.Namespace, job_manager: JobManager) -> None:
     ]
     if all(task.status() == JobStatus.COMPLETED for task in quantum_job):
         result_data: list[ResultData] = [task.result().data for task in quantum_job]
-        print(handler.poll_handler(job_data, result_data))
+        results: BenchmarkResult = handler.poll_handler(job_data, result_data)
+        print(results)
+        if is_upload:
+            upload_job(args, job_type, job_data, results, platforms[metriq_job.device_name.lower()], metriq_job.dispatch_time)
     else:
         logger.info("Job is not yet completed. Please try again later.")
 
 
+def upload_job(args: argparse.Namespace, job_type: JobType, job_data: BenchmarkData, result_data: BenchmarkResult, platform: int, dispatch_time: datetime):
```

Comment:
One option for this `upload_job` is to add a function in the `Benchmark` class, which would also support an upload pattern in addition to dispatching and polling. This would abstract the logic from `run.py` into benchmark-specific locations. We likely want to keep the the `run.py` file lean.